pkg/services/publicdashboards/queries/dashboard_queries.go

```diff
@@ -100,3 +101,12 @@ func GetDataSourceUidFromJson(query *simplejson.Json) string {
 
 	return uid
 }
+
+func RemoveMetadataFromQueryData(res *backend.QueryDataResponse) {
```

Comment:
We should probably have a unit test for this

Reply:
Yup! I was planning on adding that after the approach was confirmed 👍

Reply:
Done on https://github.com/grafana/grafana/pull/55269/commits/8b6fedf643290b0b08ad02274279349af962f59f